libs/clients/signature-collection/src/lib/signature-collection.service.ts

```diff
@@ -0,0 +1,332 @@
+      iD: parseInt(signatureId),
+    })
+    return { success: !!signature }
+  }
+
+  async removeLists(
+    collectionId: string,
+    nationalId: string,
+    listIds?: string[],
+  ): Promise<{ success: boolean }> {
+    const { id, isPresidential, isActive } = await this.currentCollectionInfo()
+    // TODO: Error mapping
+    // Lists can only be removed from current collection if it is open
+    if (id !== parseInt(collectionId) || !isActive) {
+      return { success: false }
+    }
+    // For presidentail elections remove all lists for owner, else remove selected lists
+    if (isPresidential) {
+      await this.collectionsApi.medmaelasofnunIDRemoveFrambodKennitalaPost({
+        iD: id,
+        kennitala: nationalId,
+      })
+      return { success: true }
+    }
+    if (!listIds || listIds.length === 0) {
+      return { success: false }
+    }
+    const { ownedLists } = await this.getSignee(nationalId)
+    if (!ownedLists || ownedLists.length === 0) {
+      return { success: false }
+    }
+    const listsToRemove = ownedLists.filter((list) => listIds.includes(list.id))
+    if (listsToRemove.length === 0) {
+      return { success: false }
+    }
+    // TODO: check if all where successfully removed
+    listsToRemove.map(
+      async (list) =>
+        await this.listsApi.medmaelalistarIDRemoveMedmaelalistiUserPost({
+          iD: parseInt(list.id),
+        }),
+    )
+    return { success: true }
+  }
+
+  async getUser(nationalId: string): Promise<EinstaklingurKosningInfoDTO> {
+    const { id } = await this.currentCollectionInfo()
+    return await this.collectionsApi.medmaelasofnunIDEinsInfoKennitalaGet({
+      kennitala: nationalId,
+      iD: id,
+    })
+  }
+
+  async getSignedList(nationalId: string): Promise<List | null> {
+    const { signature } = await this.getSignee(nationalId)
+    if (!signature || signature.length === 0) {
+      return null
+    }
+    // TODO: find active signature when skra updates api
+    return this.getList(signature[0].listId)
+  }
+
+  async canSign(nationalId: string): Promise<{ success: boolean }> {
+    const { canSign } = await this.getSignee(nationalId)
+    //  TODO: map errors and reasons
+    return { success: canSign }
+  }
+
+  async getSignee(nationalId: string): Promise<Signee> {
+    const { id } = await this.currentCollectionInfo()
+    const user = await this.collectionsApi.medmaelasofnunIDEinsInfoKennitalaGet(
+      {
+        kennitala: nationalId,
+        iD: id,
+      },
+    )
+
+    // TODO: find active signature when skra updates api
+    return {
+      nationalId: user.kennitala ?? '',
+      name: user.nafn ?? '',
+      electionName: user.kosningNafn ?? '',
+      canSign: user.maKjosa ?? false,
+
+      canCreate: user.maFrambod ?? false,
+      area: user.svaedi && {
+        id: user.svaedi?.id?.toString() ?? '',
+        name: user.svaedi?.nafn?.toString() ?? '',
+        min: user.svaedi?.fjoldi ?? 999,
```

Comment:
Is 999 arbitrary or a system limit?
If arbitrary, why not?
```suggestion
        min: user.svaedi?.fjoldi ?? Number.MAX_SAFE_INTEGER
```

Reply:
We don't need the min here so I removed it.